R/utilities-scenario-configuration.R

```diff
@@ -0,0 +1,115 @@
+#' Read scenario definition from excel file
+#'
+#' @details Reads scenario definition from the excel file defined in
+#' `ProjectConfiguration` and updates the `ScenarioConfiguration` with new
+#' information.
+#'
+#' @param scenarioConfiguration A `ScenarioConfiguration` that will be updated.
+#'
+#' @return Updated `ScenarioConfiguration`
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' # Create default ProjectConfiguration
+#' projectConfiguration <- createDefaultProjectConfiguration()
+#' # Create ScenarioConfiguration from ProjectConfiguration
+#' scenarioConfiguration <- ScenarioConfiguration$new(projectConfiguration)
+#' # Set scenario name
+#' scenarioConfiguration$scenarioName <- "MyScenario"
+#' # Read scenario definition from excel
+#' scenarioConfiguration <- readScenarioConfigurationFromExcel(scenarioConfiguration)
+#' }
+readScenarioConfigurationFromExcel <- function(scenarioConfiguration) {
+  # Current scenario definition structure:
+  # "Scenario_name", "IndividualId", "ModelParameterSheets", "ApplicationProtocol",
+  # "SimulationTime", "SimulationTimeUnit", "SteadyState", "ModelFile"
+  colTypes <- c("text", "text", "text", "text", "numeric", "text", "logical", "text")
```

Comment:
Is `"text"` type something we have internally designed?

Because I don't see it in _readr_: https://readr.tidyverse.org/articles/readr.html

Reply:
Its coming from the readxl package:
https://readxl.tidyverse.org/reference/read_excel.html